### DISCUSSION TITLE
How to show X-axis?

### DISCUSSION THREAD

**Discussion by pauljurczak** [alandefreitas/matplotplusplus#350] [Q&A]
My plots don't show X-axis, e.g.:

![Screenshot from 2023-06-05 20-21-27](https://github.com/alandefreitas/matplotplusplus/assets/4933032/076e5431-eaa0-4120-b9f2-683506eb0a4f)

All the examples I've seen have no X-axis either. How can I show it?

**Comment by alandefreitas** [comment]
I see the x-axis there. Its label is "Time [s]" and the ticks are [0, 5, 10, 15, 20, 25, 30].

**Reply by pauljurczak** [reply]
What I meant is an X-axis in geometrical sense, see green line below:

![Screenshot from 2023-06-05 20-21-27](https://github.com/alandefreitas/matplotplusplus/assets/4933032/94bcd5e6-e48d-42ab-96bb-4c4d2b8a0b51)

**Reply by alandefreitas** [reply]
Oh... Then you'll need to find an axis function to customize that: https://github.com/alandefreitas/matplotplusplus/blob/master/source/matplot/core/axis_type.h

I don't know if there's one, maybe https://github.com/alandefreitas/matplotplusplus/blob/289f757c26c1d93868f77ffc2fdedae862b780fa/source/matplot/core/axis_type.h#L89

as in https://github.com/search?q=repo%3Aalandefreitas%2Fmatplotplusplus%20zero_axis&type=code

At worst, you can always draw a line over the coordinate and reset the axis limits.

**Reply by pauljurczak** [reply]
> you can always draw a line

That's a good option. I found that `plt::grid(plt::on)` does an adequate job, too.
